orttraining/orttraining/core/framework/distributed_run_context.h

```diff
@@ -34,6 +39,13 @@ struct DistributedRunConfig {
   int32_t pipeline_stage_size{1};
 };
```

Comment:
What are those fields valid ranges? Are their NULL values -1?

Reply:
It ranges from 1 to world_size. And the values are validated in DistributedRunContext::DistributedRunContext()

Not sure what you mean by NULL values.. The default value is listed here and -1 should not be allowed (and will error out in the validation)

---
In reply to: [480287661](https://github.com/microsoft/onnxruntime/pull/4648#discussion_r480287661) [](ancestors = 480287661)

Reply:
Sorry. I mean ID value such as
```
  const int32_t pipeline_stage_id = config.pipeline_config.has_value() ?
                              DistributedRunContext::RankInGroup(WorkerGroupType::ModelParallel) :
                              -1;
```
but looks like they are not IDs.

---
In reply to: [480409920](https://github.com/microsoft/onnxruntime/pull/4648#discussion_r480409920) [](ancestors = 480409920,480287661)